common/containers/Tabs/Contracts/components/Interact/components/InteractForm/index.tsx

```diff
@@ -6,60 +6,76 @@ import { connect } from 'react-redux';
 import { AppState } from 'reducers';
 import { isValidETHAddress, isValidAbiJson } from 'libs/validators';
 import classnames from 'classnames';
+import Select from 'react-select';
 
-interface Props {
+interface ContractOption {
+  name: string;
+  value: string;
+}
+
+interface StateProps {
   contracts: NetworkContract[];
-  accessContract(abiJson: string, address: string): (ev) => void;
+}
+
+interface OwnProps {
+  accessContract(contractAbi: string, address: string): (ev) => void;
   resetState(): void;
 }
 
+type Props = OwnProps & StateProps;
+
 interface State {
   address: string;
   abiJson: string;
+  contract: ContractOption | null;
+  contractPlaceholder: string;
 }
 
+const abiJsonPlaceholder = [
+  {
+    type: 'constructor',
+    inputs: [{ name: 'param1', type: 'uint256', indexed: true }],
+    name: 'Event'
+  },
+  { type: 'function', inputs: [{ name: 'a', type: 'uint256' }], name: 'foo', outputs: [] }
+];
+
 class InteractForm extends Component<Props, State> {
-  public state = {
-    address: '',
-    abiJson: ''
-  };
+  private abiJsonPlaceholder = JSON.stringify(abiJsonPlaceholder, null, 0);
```

Comment:
Any reason this is a property on InteractForm?

Reply:
Not really, except for that it already was.